src/main/java/com/google/devtools/build/lib/analysis/config/transitions/ComposingTransition.java

```diff
@@ -67,6 +66,13 @@ public String getName() {
     return "(" + transition1.getName() + " + " + transition2.getName() + ")";
   }
 
+  // Override to allow recursive visiting.
+  @Override
+  public <E extends Exception> void visit(Visitor<E> visitor) throws E {
```

Comment:
How do you think this will effect my future goals of being able to look at intermediate states of composed transitions? 
On one hand, if you have starlarkTransitionA + starlarkTransitionB and starlarkTransitionA writes some gibberish that starlarkTransitionB overwrites without reading first, then we might want to throw an error upon applying starlarkTransitionA. 

On the other hand, you could argue that ComposingTransitions were never meant to be decomposed in this way (currently they're decomposed to aggregate all the effected build settings) so we shouldn't error check for that. 

That all is a digression from this particular PR, but to help understand my motives here.

Reply:
You should be able to add an override for `visit(ComposingTransition)` and handle that differently, thus letting your tree walker handle leaves and nodes separately, and getting the full structure back, if that is what you end up needing.